torntools/scripts/content/gym/ttGym.js

```diff
@@ -465,3 +468,17 @@ function setupSpecialtyGym() {
 		}
 	}
 }
+
+function displayWarning() {
+	let rawHTML, okButton;
+	if (doc.find("a#barEnergy [class^='bar-value_']").innerText.split("/")[0] > doc.find("a#barEnergy [class^='bar-value_']").innerText.split("/")[1]) {
```

Comment:
I'd say that checking the api for this one would be better.

Reply:
But it API would be slow and chain bar will always be available.

Reply:
Imo, we should use the api were possible, the html can easier be changed. An exception would be for where even a few seconds could be bad (chain timer), but otherwise the api imo.